asmcomp/amd64/emit.mlp

```diff
@@ -276,6 +277,7 @@ let call_gc_sites = ref ([] : gc_call list)
 
 let emit_call_gc gc =
   def_label gc.gc_lbl;
+  I.sub (int gc.gc_size) r15;
```

Comment:
Indeed, and this begs the question: why didn't it get caught in testing? With the current design, it's very unlikely to cause problems because the pointer is (almost) never read. I need to brainstorm with you about this.